hollow/src/main/java/com/netflix/hollow/api/consumer/metrics/AbstractRefreshMetricsListener.java

```diff
@@ -0,0 +1,178 @@
+package com.netflix.hollow.api.consumer.metrics;
+
+import com.netflix.hollow.api.consumer.HollowConsumer;
+import com.netflix.hollow.api.custom.HollowAPI;
+import com.netflix.hollow.core.read.engine.HollowReadStateEngine;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+
+public class AbstractRefreshMetricsListener extends HollowConsumer.AbstractRefreshListener  {
+
+    private long consecutiveSuccesses;
+    private long consecutiveFailures;
+
+    private Instant lastRefreshInstant;
+    private Instant refreshStart;
+
+    private ConsumerRefreshStatus refreshStatus;                    // success or failure
```

Comment:
I wonder if enums are overkill for things we can be reasonably confident will remain binary (e.g. success/fail, initial/subsequent). I've definitely gone the enum route myself in the past. I think @PaulSandoz, specifically in the case of a status enum in the producer, disliked how that went from `SUCCESS / FAIL` to `SUCCESS / FAIL / SKIP`; his preference was to stick with binary PASS / FAIL, treat skipping as passing and expose skipping through a different channel.

Reply:
sounds good